### ISSUE TITLE
cannot save TIFF

### ISSUE THREAD

**Issue by gregkotsaftis** [haraldk/TwelveMonkeys#177]
I use  v3.1.2 but cannot save TIFF images.

I have included in my path:
```
twelvemonkeys\common-image-3.1.2.jar
twelvemonkeys\common-io-3.1.2.jar
twelvemonkeys\common-lang-3.1.2.jar
twelvemonkeys\imageio-batik-3.1.2.jar
twelvemonkeys\imageio-bmp-3.1.2.jar
twelvemonkeys\imageio-core-3.1.2.jar
twelvemonkeys\imageio-jpeg-3.1.2.jar
twelvemonkeys\imageio-metadata-3.1.2.jar
twelvemonkeys\imageio-tiff-3.1.2.jar
```
I can open and view TIFF files with no problem, but when I try to save them I get:
`java.io.IOException: No appropriate writer was found for image format: tif`

**Comment by haraldk** [comment]
Hi,

Thanks for reporting!

Try changing the format to `TIFF` or `tiff`. The format name is actually TIFF (not TIF). 

But I do realize this is a little strict, so both the three and four letter versions are accepted beginning with the upcoming  3.2 release. :-)

Hope that solves the problem! If not, don't hesitate to reopen this issue. 


Best regards,

-- 
Harald K

**Comment by gregkotsaftis** [comment]
thanks, renaming the image file works; will wait for v3.2 cause scanner software just dumps *.TIF

**Comment by haraldk** [comment]
Hmmm..

Can you show me the code that you use for writing? 

Note that the file *extension* (suffix) and the *format name* are two separate things. 

For file format, the service provider (the thing that eventually instantiates the plugin) register for only `TIFF` (or lowercase `tiff`). For extension it registers for both `*.tif` and `*.tiff`.


Best regards,

-- 
Harald K

**Comment by gregkotsaftis** [comment]
the code that saves, depends on file extension.
```
BufferedImage img = ...
String format = "tif"; // determined from the original file's extension
File f = ...

if( !ImageIO.write(img, format, f) )
{
    throw( new IOException("No appropriate writer was found for image format: " + format) );
}
```

**Comment by haraldk** [comment]
I see.

Well, if you look at the JavaDoc for [ImageIO.write()](http://docs.oracle.com/javase/7/docs/api/javax/imageio/ImageIO.html#write(java.awt.image.RenderedImage,%20java.lang.String,%20java.io.File)) the parameter is really *format name*, not extension. The `File f` parameter might still be named `"foo.tif"` even if the format name is `"tiff"`. 

Anyway... This will be fixed in 3.2. :-)


Best regards,

-- 
Harald K

**Comment by gregkotsaftis** [comment]
yes, I have worked out a simple workaround like that but still I hope 3.2 won't be too late...
